### PR TITLE
Load prune configuration when a remote is created.

### PR DESCRIPTION
**Pull request by calavera**
`git_remote_create` doesn't load the prune configuration for the repository like `git_remote_lookup` does.
So `remote->prune_refs` is false after creating the remote, even when the configuration says that it should be true.

This PR changes this behaviour to load the prune configuration when a remote is created. I've added a couple of tests to verify that this works as expected.

### CODE REVIEW

File: src/remote.c [hunk 1]
```diff
@@ -454,25 +486,13 @@ int git_remote_lookup(git_remote **out, git_repository *repo, const char *name)
 			if ((error = git_config_get_bool(&remote->prune_refs, config, "fetch.prune")) < 0) {
 				if (error == GIT_ENOTFOUND) {
 					giterr_clear();
-					error = 0;
+          error = 0;
```
Comment (carlosmn) [inline src/remote.c]: Bad indentation here, it looks like

File: tests/network/fetchlocal.c [hunk 2]
```diff
@@ -481,3 +481,47 @@ void test_network_fetchlocal__call_progress(void)
...
+	git_config_free(config);
+	git_remote_free(origin);
+	git_repository_free(repo);
+}
+
+void test_network_fetchlocal__prune_load_fetch_prune_config(void)
+{
+	git_repository *repo;
+	git_remote *origin;
+  git_config *config;
```
Comment (carlosmn) [inline tests/network/fetchlocal.c]: Indentation

**Comment by carlosmn** [comment]
I think this would be the only place where we'd take `remote.<foo>.*` into consideration when creating a new remote, but whether we do that or not is basically an implementation detail, so I think it's fine.

File: src/remote.c (hunk 1 above)
Comment (calavera) [inline src/remote.c]: fuuu, I though I had fix all of those.

**Comment by calavera** [comment]
@carlosmn I've fixed those indentation mistakes, thanks for pointing them out.

**Comment by carlosmn** [comment]
:+1: we should get this in for the next rc

File: tests/network/fetchlocal.c [hunk 3]
```diff
@@ -481,3 +481,47 @@ void test_network_fetchlocal__call_progress(void)
 	git_remote_free(remote);
 	git_repository_free(repo);
 }
+
+void test_network_fetchlocal__prune_load_remote_prune_config(void)
+{
+	git_repository *repo;
+	git_remote *origin;
+  git_config *config;
```
Comment (ethomson) [inline tests/network/fetchlocal.c]: Sorry to nitpick, one more whitespace issue here, then I agree that this is ready for merging.

**Comment by calavera** [comment]
Thanks for letting me know @ethomson. It should be fixed now.

**Comment by ethomson** [comment]
Thanks!

**Pull request merged**